aiida/cmdline/params/types/plugin.py

```diff
@@ -68,49 +67,52 @@ def __init__(self, group=None, load=False, *args, **kwargs):
         """
         # pylint: disable=keyword-arg-before-vararg
+        self.load = load
+        self._input_group = group
+
+        super().__init__(*args, **kwargs)
+
+    def _get_valid_groups(self):
+        """Get allowed groups for this instance"""
+
+        group = self._input_group
         valid_entry_point_groups = get_entry_point_groups()
 
         if group is None:
-            self._groups = tuple(valid_entry_point_groups)
+            return tuple(valid_entry_point_groups)
+
+        if isinstance(group, str):
+            unvalidated_groups = (group,)
+        elif isinstance(group, tuple):
+            unvalidated_groups = group
         else:
-            if isinstance(group, str):
-                invalidated_groups = tuple([group])
-            elif isinstance(group, tuple):
-                invalidated_groups = group
-            else:
-                raise ValueError('invalid type for group')
+            raise ValueError('invalid type for group')
 
-            groups = []
+        groups = []
 
-            for grp in invalidated_groups:
+        for grp in unvalidated_groups:
 
-                if not grp.startswith(ENTRY_POINT_GROUP_PREFIX):
-                    grp = ENTRY_POINT_GROUP_PREFIX + grp
+            if not grp.startswith(ENTRY_POINT_GROUP_PREFIX):
+                grp = ENTRY_POINT_GROUP_PREFIX + grp
 
-                if grp not in valid_entry_point_groups:
-                    raise ValueError(f'entry point group {grp} is not recognized')
+            if grp not in valid_entry_point_groups:
+                raise ValueError(f'entry point group {grp} is not recognized')
 
-                groups.append(grp)
+            groups.append(grp)
 
-            self._groups = tuple(groups)
+        return tuple(groups)
 
-        self._init_entry_points()
-        self.load = load
+    @functools.cached_property
+    def groups(self) -> tuple:
+        return self._get_valid_groups()
 
-        super().__init__(*args, **kwargs)
-
-    def _init_entry_points(self):
-        """
-        Populate entry point information that will be used later on.  This should only be called
-        once in the constructor after setting self.groups because the groups should not be changed
-        after instantiation
-        """
-        self._entry_points = [(group, entry_point) for group in self.groups for entry_point in get_entry_points(group)]
-        self._entry_point_names = [entry_point.name for group in self.groups for entry_point in get_entry_points(group)]
+    @functools.cached_property
+    def _entry_points(self) -> list[tuple]:
+        return [(group, entry_point) for group in self.groups for entry_point in get_entry_points(group)]
 
-    @property
-    def groups(self):
-        return self._groups
+    @functools.cached_property
+    def _entry_point_names(self) -> list[str]:
+        return [entry_point.name for group in self.groups for entry_point in get_entry_points(group)]
```

Comment:
I think this should be a slight performance increase right, relying only on `_entry_points_names` to do the double loop and `get_entry_points` call?
```suggestion
        return [entry_point.name for _, entry_point in self._entry_points]
```

Reply:
Good catch, done.